docs/mempools/dd.md

```diff
@@ -0,0 +1,106 @@
+# Limiting Client Mempools Design Document
+
+Our client tx mempool today grows unbounded. We need to place a limit on how many txs it absorbs.
+
+This addresses how to limit the total number of valid txs. It does **not** address how to protect against DOS attacks that cause the client to validate more txs than it can process.
+
+## Tx validity
+
+A tx in Aztec requires the following checks to be valid. Static checks can be done just once, dynamic ones are done when the tx is received and again when the sequencer is building a block. If the tx fails validation during block building, it gets evicted (unless it failed due to _Gas fees over block base fee_).
+
+### Static
+
+- Chain id
+- Setup function
+- Correct public execution requests and logs
+- Minimum gas fees (base and priority)
+- Reasonable gas limit (currently missing)
+
+### Dynamic
+
+- Max block number for inclusion
+- Double spend (repeated nullifiers)
+- Archive root exists (can become invalid after a reorg)
+- Fee juice balance for fee payer
+- Gas fees over current block base fee
+- Gas limit below current block target
```

Comment:
We are having a target and limit similarly to 1559, so I think we would allow as long as it below block limit and not "just" the target.